packages/app-desktop/gui/JoplinCloudConfigScreen.tsx

```diff
@@ -0,0 +1,65 @@
+const { connect } = require('react-redux');
+import { AppState } from '../app.reducer';
+import { _ } from '@joplin/lib/locale';
+import { clipboard } from 'electron';
+import Setting from '@joplin/lib/models/Setting';
+import { reg } from '@joplin/lib/registry';
+import useAsyncEffect, { AsyncEffectEvent } from '@joplin/lib/hooks/useAsyncEffect';
+import Button from './Button/Button';
+
+type JoplinCloudConfigScreenProps = {
+	inboxEmail: string;
+	syncTarget: number;
+	joplinCloud: {
+		username: string;
+		password: string;
+		baseUrl: string;
+		userContentPath: string;
+	};
+};
+
+const JoplinCloudConfigScreen = (props: JoplinCloudConfigScreenProps) => {
+	const copyToClipboard = () => {
+		clipboard.writeText(props.inboxEmail);
+	};
+
+	useAsyncEffect(async (event: AsyncEffectEvent) => {
+		if (props.inboxEmail !== '') return;
+
+		const syncTarget = reg.syncTarget();
+		const fileApi = await syncTarget.fileApi();
+		const api = fileApi.driver().api();
+
+		const owner = await api.exec('GET', `api/users/${api.userId}`);
+
+		if (event.cancelled) return;
+
+		if (owner.inbox_email) {
+			Setting.setValue('emailToNote.inboxEmail', owner.inbox_email);
+		}
+	}, [props.inboxEmail, reg, Setting]);
```

Comment:
In https://github.com/laurent22/joplin/pull/8461/files you have some code to fetch the inbox email. I think you should call that here

Reply:
While the request is the same, the logic before and after is a little different (in one we are setting the `inbox_email` and the other we are setting the `inbox.jop_id`)

Maybe we could have only one function like so, what do you think?

```js
		if (
                     Setting.value('emailToNote.inboxEmail') !== '' &&
                     Setting.value('emailToNote.inboxJopId') !== ''
                ) return;

		const syncTarget = reg.syncTarget();
		const fileApi = await syncTarget.fileApi();
		const api = fileApi.driver().api();

		const owner = await api.exec('GET', `api/users/${api.userId}`);

		if (owner.inbox_email) {
			Setting.setValue('emailToNote.inboxEmail', owner.inbox_email);
		}

		if (owner.inbox) {
			Setting.setValue('emailToNote.inboxJopId', owner.inbox.jop_id);
		}
```

Reply:
I agree, that would make sense to set both at the same time